test/test_client.py

```diff
@@ -142,31 +146,26 @@ def test_handshake_extra_accept_headers():
 
 @pytest.mark.parametrize("extra_headers", [[], [(b"connection", b"Keep-Alive")]])
 def test_handshake_response_broken_connection_header(extra_headers):
-    events = _make_handshake(101, [(b"upgrade", b"WebSocket")] + extra_headers)
-    assert events == [
-        Fail(
-            code=CloseReason.PROTOCOL_ERROR, reason="Missing Connection: Upgrade header"
-        )
-    ]
+    with pytest.raises(RemoteProtocolError) as excinfo:
+        events = _make_handshake(101, [(b"upgrade", b"WebSocket")] + extra_headers)
+    assert str(excinfo.value) == "Missing Connection: Upgrade header"
```

Comment:
This message is a bit confusing with the colon and key-value pair.
Maybe phrase it like: `Missing header! Connection: upgrade`
(same below)